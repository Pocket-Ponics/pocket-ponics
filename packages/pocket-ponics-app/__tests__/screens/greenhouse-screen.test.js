import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity } from 'react-native'

import GreenhouseScreen from '../../screens/greenhouse-screen'
import GreenhouseList from '../../components/greenhouse/greenhouse-list'

describe('Greenhouse Screen', () => {
	test('renders correctly', () => {
		global.greenhouses = {}
		const tree = shallow(
			<GreenhouseScreen navigation={{ 
				getParam: jest.fn(), 
				addListener: jest.fn().mockImplementationOnce((str, fn) => fn()) 
			}} />
		).html()
		expect(tree).toMatchSnapshot()
	})

	test('renders with a specific greenhouse', () => {
		global.greenhouses = { 7: {} }
		const tree = shallow(
			<GreenhouseScreen navigation={{ 
				getParam: jest.fn().mockReturnValue(7), 
				addListener: jest.fn().mockImplementationOnce((str, fn) => fn()) 
			}} />
		).html()
		expect(tree).toMatchSnapshot()
	})

	test('swapIndex changes greenhouse', () => {
		global.greenhouses = {}
		const tree = shallow(
			<GreenhouseScreen navigation={{ 
				getParam: jest.fn(), 
				addListener: jest.fn().mockImplementationOnce((str, fn) => fn()),
				setParams: jest.fn()
			}} />
		)

		tree.find(GreenhouseList).at(0).props().swapItem(0)
		expect(tree.html()).toMatchSnapshot()
	})

	test('navigation options display profile button', () => {
		const ProfileButton = GreenhouseScreen.navigationOptions({ navigation: {
			navigate: jest.fn()
		}}).headerRight
		const tree = shallow(
			<ProfileButton />
		)

		tree.find(TouchableOpacity).at(0).props().onPress(0)
		expect(tree.html()).toMatchSnapshot()
	})
})