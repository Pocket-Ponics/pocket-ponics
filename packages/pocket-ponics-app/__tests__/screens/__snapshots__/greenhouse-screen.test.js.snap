// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greenhouse Screen navigation options display profile button 1`] = `"<View style=\\"padding:10px;opacity:1\\"><Text style=\\"font-weight:bold;color:#FFFFFF;font-size:15px\\">Profile</Text></View>"`;

exports[`Greenhouse Screen renders correctly 1`] = `"<View style=\\"flex:1;background-color:#472600\\"><RCTScrollView decelerationRate=\\"fast\\" overScrollMode=\\"never\\" initialNumToRender=\\"6\\" maxToRenderPerBatch=\\"13\\" windowSize=\\"13\\" data=\\"[object Object]\\" firstItem=\\"0\\" itemWidth=\\"750\\" style=\\"opacity:0;width:750px;flex-direction:row\\" sliderWidth=\\"750\\" inactiveSlideScale=\\"0.95\\" inactiveSlideOpacity=\\"1\\" activeSlideAlignment=\\"start\\" activeAnimationType=\\"spring\\" activeAnimationOptions=\\"[object Object]\\" activeSlideOffset=\\"20\\" apparitionDelay=\\"0\\" autoplayDelay=\\"1000\\" autoplayInterval=\\"3000\\" callbackOffsetMargin=\\"5\\" containerCustomStyle=\\"[object Object]\\" contentContainerCustomStyle=\\"[object Object]\\" inactiveSlideShift=\\"0\\" layout=\\"default\\" lockScrollTimeoutDuration=\\"1000\\" loopClonesPerSide=\\"3\\" slideStyle=\\"[object Object]\\" swipeThreshold=\\"20\\" contentContainerStyle=\\"[object Object],[object Object]\\" scrollEventThrottle=\\"1\\" numColumns=\\"1\\" updateCellsBatchingPeriod=\\"50\\" viewabilityConfigCallbackPairs=\\"\\" stickyHeaderIndices=\\"\\"><View><View style=\\"0:[object Object]\\"></View></View></RCTScrollView></View>"`;

exports[`Greenhouse Screen renders with a specific greenhouse 1`] = `"<View style=\\"flex:1;background-color:#472600\\"><View style=\\"align-content:center\\"><View style=\\"flex-direction:row;align-items:center;justify-content:center\\"><View style=\\"opacity:1\\"><Text style=\\"font-size:25px;padding-left:5px;padding-right:5px;font-weight:bold;color:#FFFFFF\\">â€¢</Text></View><View style=\\"opacity:1\\"><Text style=\\"font-size:25px;padding-left:5px;padding-right:5px;font-weight:bold;color:#FFFFFF;opacity:0.7\\">+</Text></View></View></View><RCTScrollView decelerationRate=\\"fast\\" overScrollMode=\\"never\\" initialNumToRender=\\"6\\" maxToRenderPerBatch=\\"13\\" windowSize=\\"13\\" data=\\"[object Object],[object Object]\\" firstItem=\\"0\\" itemWidth=\\"750\\" style=\\"opacity:0;width:750px;flex-direction:row\\" sliderWidth=\\"750\\" inactiveSlideScale=\\"0.95\\" inactiveSlideOpacity=\\"1\\" activeSlideAlignment=\\"start\\" activeAnimationType=\\"spring\\" activeAnimationOptions=\\"[object Object]\\" activeSlideOffset=\\"20\\" apparitionDelay=\\"0\\" autoplayDelay=\\"1000\\" autoplayInterval=\\"3000\\" callbackOffsetMargin=\\"5\\" containerCustomStyle=\\"[object Object]\\" contentContainerCustomStyle=\\"[object Object]\\" inactiveSlideShift=\\"0\\" layout=\\"default\\" lockScrollTimeoutDuration=\\"1000\\" loopClonesPerSide=\\"3\\" slideStyle=\\"[object Object]\\" swipeThreshold=\\"20\\" contentContainerStyle=\\"[object Object],[object Object]\\" scrollEventThrottle=\\"1\\" numColumns=\\"1\\" updateCellsBatchingPeriod=\\"50\\" viewabilityConfigCallbackPairs=\\"\\" stickyHeaderIndices=\\"\\"><View><View style=\\"0:[object Object]\\"></View><View style=\\"0:[object Object]\\"></View></View></RCTScrollView></View>"`;

exports[`Greenhouse Screen swapIndex changes greenhouse 1`] = `"<View style=\\"flex:1;background-color:#472600\\"><RCTScrollView decelerationRate=\\"fast\\" overScrollMode=\\"never\\" initialNumToRender=\\"6\\" maxToRenderPerBatch=\\"13\\" windowSize=\\"13\\" data=\\"[object Object]\\" firstItem=\\"0\\" itemWidth=\\"750\\" style=\\"opacity:0;width:750px;flex-direction:row\\" sliderWidth=\\"750\\" inactiveSlideScale=\\"0.95\\" inactiveSlideOpacity=\\"1\\" activeSlideAlignment=\\"start\\" activeAnimationType=\\"spring\\" activeAnimationOptions=\\"[object Object]\\" activeSlideOffset=\\"20\\" apparitionDelay=\\"0\\" autoplayDelay=\\"1000\\" autoplayInterval=\\"3000\\" callbackOffsetMargin=\\"5\\" containerCustomStyle=\\"[object Object]\\" contentContainerCustomStyle=\\"[object Object]\\" inactiveSlideShift=\\"0\\" layout=\\"default\\" lockScrollTimeoutDuration=\\"1000\\" loopClonesPerSide=\\"3\\" slideStyle=\\"[object Object]\\" swipeThreshold=\\"20\\" contentContainerStyle=\\"[object Object],[object Object]\\" scrollEventThrottle=\\"1\\" numColumns=\\"1\\" updateCellsBatchingPeriod=\\"50\\" viewabilityConfigCallbackPairs=\\"\\" stickyHeaderIndices=\\"\\"><View><View style=\\"0:[object Object]\\"></View></View></RCTScrollView></View>"`;
